airbyte-integrations/connectors/source-intercom/source_intercom/source.py

```diff
@@ -280,10 +286,12 @@ class Conversations(IncrementalIntercomStream):
 
     data_fields = ["conversations"]
 
-    def request_params(self, next_page_token: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
-        params = super().request_params(next_page_token, **kwargs)
-        params.update({"order": "asc", "sort": self.cursor_field})
-        return params
+    # We're sorting by desc. Once we hit the first page with an out-of-date result we can stop.
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        if self.has_old_records:
+            return None
+
+        return super().next_page_token(response)
```

Comment:
To what extent this can also be applied to other streams? Could we avoid unnecessary requests on other streams too?

Reply:
Frustratingly enough it looks like Conversations is the only list item that has an explicit sort, so while some models may incidentally support this, it's hard to be confident since the API makes no sort guarantees on any other list endpoint.